_posts/2021-11-03-flink-backward.md

```diff
@@ -0,0 +1,79 @@
+
+We looked at the origin of these emotions and came up with an analysis of what went well and what could be improved. We also incorporated some feedback gathered from the community.
+
+## Problems faced
+
+From a content perspective, the community is still ironing out processes around documentation and blog posts. There have been a lot of test instabilities and issues with some of the efforts to fix them. We also had to push the feature freeze by two weeks, which might actually be considered early and not actually be living up to the Apache Flink release tradition. :p
+
+## Things enjoyed
+
+The implementation of some features, such as [buffer debloating](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/memory/network_mem_tuning/#the-buffer-debloating-mechanism) and [fine-grained resource management](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/), went really smoothly. Though a few issues are now popping up. We enjoyed the moments while they last. :smiling_face_with_tear: 
+
+We also said goodbye to some components. It always feels refreshing to remove code and reduce complexity. :slightly_smiling_face:
+
+
+# What we want to achieve through process changes
+
+## Transparency - let the community participate
+
+When approaching a release (usually a couple of weeks after the previous release has been done) we set up bi-weekly meetings for the community to discuss any issues regarding the release. The usefulness of those meetings varied a lot and so we started to [track the efforts](https://cwiki.apache.org/confluence/display/FLINK/1.14+Release) in the Apache Flink Confluence wiki. 
+
+We came up with a system to label the current states of each feature: "independent", "won’t make it", "very unlikely", "will make it", "done", and "done done". We introduced the "done done" state since we were lacking a shared understanding of the definition of done. To qualify for "done done", the feature is manually tested by someone who has not been involved in the implementation of the specific effort and there exists comprehensive documentation that enables users to use the feature.
+
+After each meeting, we provided updates on the mailing list and created a corresponding burn down chart. Those efforts have been perceived positively although they might still require some improvements.
+
+The meeting used to only be for those who have been driving the main efforts, but we opened it up to the whole community for this release. Nobody ended up joining :smiling_face_with_tear: but we will continue to make the meetings open to everyone.   
+
+
+## Stability - reduce building and testing pain
+
+At one point as we were coming close to the feature freeze, the stability of the master branch became quite unstable. Although we have encountered this issue in the past, building and testing Flink under such conditions was not ideal. Let's not count how often Kafka integrations tests have failed. 
```

Comment:
I don't like the Kafka comment, I understand it's a joke it doesn't come across well. We can certainly talk about why Kafka is difficult to test against but that feel separate from this. 

```suggestion
At one point, as we were coming close to the feature freeze, the stability of the master branch became quite unstable. Although we have encountered this issue in the past, building and testing Flink under such conditions was not ideal.
```